catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/GeoPdfParser.java

```diff
@@ -0,0 +1,298 @@
+     * in this fashion.
+     * @param pdPage the page to generate the NeatLine
+     * @return an array of points representing the NeatLine
+     */
+    private COSArray generateNeatLineFromPDFDimensions(PDPage pdPage) {
+        COSArray neatLineArray = new COSArray();
+
+        String width = pdPage.getMediaBox().getWidth() + "";
+        String height = pdPage.getMediaBox().getHeight() + "";
+
+        neatLineArray.add(new COSString("0"));
+        neatLineArray.add(new COSString("0"));
+
+        neatLineArray.add(new COSString(width));
+        neatLineArray.add(new COSString("0"));
+
+        neatLineArray.add(new COSString(width));
+        neatLineArray.add(new COSString(height));
+
+        neatLineArray.add(new COSString("0"));
+        neatLineArray.add(new COSString(height));
+
+        return neatLineArray;
+    }
+
+    /**
+     * Convert a Point2d into WKT Lat/Lon
+     *
+     * @param point2D
+     * @return a String representation of a WKT Lat/Lon pair
+     */
+    private String point2dToWkt(Point2D point2D) {
+        return point2D.getX() + " " + point2D.getY();
+    }
+
+    /**
+     * Parses a given NeatLine and Transformation matrix into a WKT String
+     *
+     * @param lgidict         - The PDF's LGIDict object
+     * @param neatLineArray   - The NeatLine array of points for the PDF
+     * @param toDoubleVisitor - A visitor that converts PDF Strings / Ints / Longs into doubles.
+     * @return the generated WKT Lat/Lon set
+     * @throws IOException
+     */
+    private String getWktFromNeatLine(COSDictionary lgidict, COSArray neatLineArray,
+            ICOSVisitor toDoubleVisitor) throws IOException {
+        List<Double> neatline = new LinkedList<>();
+        StringBuilder wktString = new StringBuilder();
+        List<String> coordinateList = new LinkedList<>();
+        String firstCoordinate = null;
+
+        double[] points = new double[CTM_SIZE];
+        for (int i = 0; i < CTM_SIZE; i++) {
+            points[i] = (Double) lgidict.getObjectFromPath(CTM + "/[" + i + "]")
+                    .accept(toDoubleVisitor);
+        }
+        AffineTransform affineTransform = new AffineTransform(points);
+
+        for (int i = 0; i < neatLineArray.size(); i++) {
+            neatline.add((Double) neatLineArray.get(i)
+                    .accept(toDoubleVisitor));
+        }
+
+        for (int i = 0; i < neatline.size(); i += 2) {
+            double x = neatline.get(i);
+            double y = neatline.get(i + 1);
+
+            Point2D p = new Point2D.Double(x, y);
+
+            Point2D pprime = affineTransform.transform(p, null);
+
+            String xySet = point2dToWkt(pprime);
+
+            if (firstCoordinate == null) {
+                firstCoordinate = xySet;
+            }
+            coordinateList.add(xySet);
+        }
+        coordinateList.add(firstCoordinate);
+        wktString.append(StringUtils.join(coordinateList, ", "));
```

Comment:
Looks like `wktString` doesn't need to be a `StringBuilder` since `append()` is only called once.